lib/Hpe3parSdk/remote_copy_manager.rb

```diff
@@ -0,0 +1,293 @@
+        raise Hpe3parSdk::HPE3PARException(ex.message)
+      end
+      response
+    end
+
+    def start_rcopy()
+      begin
+        cmd = ['startrcopy']
+        response = @ssh.run(cmd)
+        if response != []
+          raise Hpe3parSdk::HPE3PARException(message: response)
+        end
+      rescue Hpe3parSdk::HPE3PARException => ex
+        raise Hpe3parSdk::HPE3PARException(ex.message)
+      end
+      response
+    end
+
+    def rcopy_service_exists()
+      cmd = ['showrcopy']
+      response = @ssh.run(cmd)
+      rcopyservice_status = false
+      if response[2].include?('Started')
+        rcopyservice_status = true
+      end
+      rcopyservice_status
+    end
+
+    def get_remote_copy_link(link_name)
+      response, body = @http.get('/remotecopylinks/#{link_name}')
+      body
+    end
+
+    def rcopy_link_exists(targetName, local_port, target_system_peer_port)
+      rcopylink_exits = false
+      link_name = targetName + '_' + local_port.replace(':', '_')
+      begin
+        response = get_remote_copy_link(link_name)
+        if response and response['address'] == target_system_peer_port
+          rcopylink_exits = true
+        end
+      rescue Hpe3parSdk::HTTPNotFound => ex
+        pass
+      end
+      rcopylink_exits
+    end
+
+    def admit_remote_copy_target( targetName, mode, remote_copy_group_name,
+                              source_target_volume_pairs_list=[])
+      if source_target_volume_pairs_list == []
+        cmd = ['admitrcopytarget', targetName, mode, remote_copy_group_name]
+      else
+        cmd = ['admitrcopytarget', targetName, mode, remote_copy_group_name]
+        for volume_pair_tuple in source_target_volume_pairs_list
+          source_target_pair = volume_pair_tuple[0] + ':' + volume_pair_tuple[1]
+          cmd << source_target_pair
+        end
+      end
+      begin
+        command = cmd.join(" ")
+        response = @ssh.run(command)
+        if response != []
+          raise Hpe3parSdk::HPE3PARException(message: response)
+        end
+      rescue Hpe3parSdk::HPE3PARException => ex
+        raise Hpe3parSdk::HPE3PARException(ex.message)
+      end
+      response
+    end
+
+    def dismiss_remote_copy_target( targetName, remote_copy_group_name)
+      option = '-f'
+      cmd = ['dismissrcopytarget', option, targetName, remote_copy_group_name]
+      begin
+        command = cmd.join(" ")
+        response = @ssh.run(command)
+        if response != []
+          raise Hpe3parSdk::HPE3PARException(message: response)
+        end
+      rescue Hpe3parSdk::HPE3PARException => ex
+        raise Hpe3parSdk::HPE3PARException(ex.message)
+      end
+      response
+    end
+
+    def target_in_remote_copy_group_exists( target_name, remote_copy_group_name)
+      begin
+        contents = self.get_remote_copy_group(remote_copy_group_name)
+        for item in contents['targets']
+          if item['target'] == target_name
+            return true           
```

Comment:
Are we returning true for all items? or we need to do it once all the items are iterated?